File: Resoluções-das-Listas/Lista-1-Problema-1.py
import math

g = 9.81

v0 = float(input("Digite a velocidade inicial (m/s): "))
theta = float(input("Digite o ângulo inicial (graus): "))

theta_rad = math.radians(theta)

h_max = (v0 ** 2 * (math.sin(theta_rad) ** 2)) / (2 * g)

alcance = (v0 ** 2 * math.sin(2 * theta_rad)) / g

tempo_total = (2 * v0 * math.sin(theta_rad)) / g

print(f"Altura máxima: {h_max} m")
print(f"Alcance horizontal: {alcance} m")
print(f"Tempo de movimento: {tempo_total} s")

with open("trajetoria.txt", "w") as arquivo:
    t = 0.0
    dt = 0.01
    while t <= tempo_total:
        x = v0 * math.cos(theta_rad) * t
        y = v0 * math.sin(theta_rad) * t - 0.5 * g * t ** 2
        arquivo.write(f"{x} {y}\n")
        t += dt

print("Trajetória salva em 'trajetoria.txt'")

File: Resoluções-das-Listas/Lista-1-Problema-1.cpp
#include <iostream>
#include <fstream>
#include <cmath>

const double g = 9.81;

using namespace std;

int main()
{
    double v0, theta;

    cout << "Digite a velocidade inicial (m/s): ";
    cin >> v0;

    cout << "Digite o ângulo inicial (graus): ";
    cin >> theta;

    double theta_rad = theta * M_PI / 180.0;

    double h_max = (v0 * v0 * pow(sin(theta_rad), 2)) / (2 * g);

    double alcance = (v0 * v0 * sin(2 * theta_rad)) / g;

    double tempo_total = (2 * v0 * sin(theta_rad)) / g;

    cout << "Altura máxima: " << h_max << " m" << std::endl;
    cout << "Alcance horizontal: " << alcance << " m" << std::endl;
    cout << "Tempo de movimento: " << tempo_total << " s" << std::endl;

    ofstream arquivo("trajetoria.txt");
    if (arquivo.is_open())
    {
        double t = 0.0;
        double dt = 0.01;
        while (t <= tempo_total)
        {
            double x = v0 * cos(theta_rad) * t;
            double y = v0 * sin(theta_rad) * t - 0.5 * g * t * t;
            arquivo << x << " " << y << std::endl;
            t += dt;
        }
        arquivo.close();
        std::cout << "Trajetória salva em 'trajetoria.txt'" << std::endl;
    }
    else
    {
        std::cerr << "Erro ao abrir o arquivo para escrita." << std::endl;
    }

    return 0;
}